source/application/DbgHelp/DbgHelpStackWalkCallback.h

```diff
@@ -25,44 +27,38 @@ class DbgHelpStackWalkCallback : public IDbgHelpStackWalkCallback
 public:
     DbgHelpStackWalkCallback(HANDLE process, IDiaStackFrame* frame, const DbgHelpPrivate::ModuleData* mod_info);
 
-    const std::wstring GetModuleName();
-    const std::wstring GetFunctionName();
-    const DWORD GetParameterCount();
-    DbgHelpArgType GetParameterType(DWORD num);
-    std::wstring GetParameterTypeName(DWORD num);
-    std::wstring GetParameterName(DWORD num);
-    std::shared_ptr<void> GetParameterValue(DWORD num);
-    std::wstring GetFunctionParameters(); // TODO: Destroy
+    std::wstring GetModuleName() override;
+    std::wstring GetFunctionName() override;
+    std::vector<Parameter> GetParameters() override;
 
     DWORD GetUnsureStatus();
 private:
     /*
-     * These functions are just helper methods and will do very little data validation
+     * TODO: Cache more?
+     * -- Warepire
+     */
+    struct ParamInfo
+    {
+        DbgHelpType m_type;
+        std::optional<std::wstring> m_name;
+        CComPtr<IDiaSymbol> m_arg_info;
+        CComPtr<IDiaSymbol> m_type_info;
+        size_t m_offset;
+    };
+
+    /*
+     * These functions are just helper methods and will do very little data validation.
      * Always validate data before calling them.
      */
     CComPtr<IDiaSymbol> GetFunctionSymbol();
     ULONGLONG GetProgramCounter();
     void EnumerateParameters();
+    std::optional<ParameterValue> GetParameterValue(const ParamInfo& param_info);
 
     HANDLE m_process;
     IDiaStackFrame* m_frame;
     const DbgHelpPrivate::ModuleData* m_mod_info;
 
-    /*
-     * TODO: Cache more?
-     * -- Warepire
-     */
-    struct ParamInfo
-    {
-        DbgHelpArgType m_type;
-        std::wstring m_typename;

```

Comment:
Now I remember what this was for, UDTs and Enum types, as they carry a name.

For example "ParamInfo".